File: java/run-length-encoding/src/main/java/RunLengthEncoding.java
import java.util.regex.Pattern;

class RunLengthEncoding {

    String encode(String data) {
        return Pattern.compile("(.)\\1+").matcher(data)
                .replaceAll(m -> String.format("%d%s", m.group().length(), m.group().charAt(0)));
    }

    String decode(String data) {
        return Pattern.compile("(\\d+)(.)").matcher(data)
                .replaceAll(m -> m.group(2).repeat(Integer.parseInt(m.group(1))));
    }

}

File: python/run-length-encoding/run_length_encoding.py
import re


def decode(string):
    return re.sub(r'(\d+)(.)', lambda m: m.group(2) * int(m.group(1)), string)


def encode(string):
    return re.sub(r'(.)\1+', lambda m: f'{len(m.group(0))}{m.group(1)}', string)
